File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/fields/NameField.java
package com.saizad.mvvm.components.form.fields;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.sa.easyandroidfrom.StringUtils;
import com.sa.easyandroidfrom.fields.NonEmptyStringField;
import com.saizad.mvvm.components.form.FieldValidatorKt;

import javax.annotation.Nonnull;


public class NameField extends NonEmptyStringField {

    public NameField(@Nonnull String fieldId, @NonNull String firstName, @NonNull String lastName) {
        this(fieldId, firstName + " " + lastName);
    }

    public NameField(@NonNull String fieldId, @Nullable String ogField) {
        this(fieldId, ogField, true);
    }

    public NameField(@NonNull String fieldId, @Nullable String ogField, boolean isMandatory) {
        super(fieldId, ogField, isMandatory);
    }

    public NameField(@Nonnull String fieldId) {
        super(fieldId, null, true);
    }

    @Nullable
    public String getFirstName() {
        return getName(0);
    }

    @Nullable
    public String getLastName() {
        return getName(1);
    }

    @Nullable
    private String getName(int index) {
        if (isFieldValid()) {
            final String[] split = StringUtils.stripTrailingLeadingNewLines(getField()).split(" ");
            return split[index];
        }
        return null;
    }

    @Override
    public void validate() throws Exception {
        FieldValidatorKt.validateName(getField());
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/fields/NameField.kt
package com.saizad.mvvm.components.form.fields

import com.sa.easyandroidform.StringUtils
import com.sa.easyandroidform.fields.NonEmptyStringField
import com.saizad.mvvm.components.form.validateName
import io.reactivex.exceptions.CompositeException
import kotlin.jvm.Throws

class NameField(
    fieldId: String,
    val name: String,
    isMandatory: Boolean
) : NonEmptyStringField(fieldId, name, isMandatory) {

    fun firstName(): String? = getName(0)

    fun lastName(): String? = getName(1)

    private fun getName(index: Int): String? {
        if (isValid) {
            val split =
                StringUtils.stripTrailingLeadingNewLines(field)
                    .split(" ".toRegex()).toTypedArray()
            return split[index]
        }
        return null
    }

    @Throws(CompositeException::class)
    override fun validate() {
        super.validate()
        validateName(field)
    }
}